dev/com.ibm.ws.webserver.plugin.runtime/src/com/ibm/ws/webserver/plugin/runtime/listeners/GeneratePluginConfigListener.java

```diff
@@ -67,6 +67,8 @@ public class GeneratePluginConfigListener implements RuntimeUpdateListener, Appl
     private WsLocationAdmin locationService;
 
     private static GeneratePluginConfigListener theListener = null;
+    private static final String CHECKPOINT_GEN_PLUGIN = "io.openliberty.checkpoint.generate.plugin";
+    private BundleContext bc;
```

Comment:
Instead of storing `bc` as an instance variable I would store
```java
private boolean generateOnRestore;
```
Then in `activate`
```java
    generateOnRestore=Boolean.valueOf(bc == null ? null : bc.getProperty(CHECKPOINT_GEN_PLUGIN));
```

Then use the `generateOnRestore` in the code.